backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/components/DiagramComponentProps.java

```diff
@@ -58,4 +63,78 @@ public DiagramDescription getDiagramDescription() {
     public List<ViewCreationRequest> getViewCreationRequests() {
         return this.viewCreationRequests;
     }
+
+    public MoveEvent getMoveEvent() {
+        return this.moveEvent;
+    }
+
+    public Position getStartingPosition() {
+        return this.startingPosition;
+    }
+
+    public static Builder newDiagramComponentProps() {
+        return new Builder();
+    }
+
+    /**
+     * The Builder to create a new {@link DiagramComponentProps}.
+     *
+     * @author fbarbin
+     */
+    @SuppressWarnings("checkstyle:HiddenField")
+    public static final class Builder {
+        private VariableManager variableManager;
+
+        private DiagramDescription diagramDescription;
+
+        private Optional<Diagram> previousDiagram;
+
+        private List<ViewCreationRequest> viewCreationRequests;
+
+        private MoveEvent moveEvent;
+
+        private Position startingPosition;
+
+        public Builder variableManager(VariableManager variableManager) {
+            this.variableManager = Objects.requireNonNull(variableManager);
+            return this;
+        }
+
+        public Builder diagramDescription(DiagramDescription diagramDescription) {
+            this.diagramDescription = Objects.requireNonNull(diagramDescription);
+            return this;
+        }
+
+        public Builder previousDiagram(Optional<Diagram> previousDiagram) {
+            this.previousDiagram = Objects.requireNonNull(previousDiagram);
+            return this;
+        }
+
+        public Builder viewCreationRequests(List<ViewCreationRequest> viewCreationRequests) {
+            this.viewCreationRequests = Objects.requireNonNull(viewCreationRequests);
+            return this;
+        }
+
+        public Builder moveEvent(MoveEvent moveEvent) {
+            this.moveEvent = Objects.requireNonNull(moveEvent);
+            return this;
+        }
+
+        public Builder startingPosition(Position startingPosition) {
+            this.startingPosition = Objects.requireNonNull(startingPosition);
+            return this;
+        }
+
+        public DiagramComponentProps build() {
+            DiagramComponentProps diagramComponentProps = new DiagramComponentProps();
+            diagramComponentProps.variableManager = Objects.requireNonNull(this.variableManager);
+            diagramComponentProps.diagramDescription = Objects.requireNonNull(this.diagramDescription);
+            diagramComponentProps.previousDiagram = Objects.requireNonNull(this.previousDiagram);
+            diagramComponentProps.viewCreationRequests = List.copyOf(Objects.requireNonNull(this.viewCreationRequests));
+            diagramComponentProps.moveEvent = this.moveEvent;
+            diagramComponentProps.startingPosition = this.startingPosition;
```

Comment:
Missing checks for `Objects.requireNonNull`? Or of these are optional maybe that should be made explicit by having `Optional<MoveEvent> getMoveEvent()` in `DiagramComponentProps` and `diagramComponentProps.moveEvent = Optional.ofNullable(this.moveEvent);` here.

Reply:
Indeed, those values are optional. Done